src/loadgenerator/locustfile.py

```diff
@@ -50,11 +50,12 @@ def signup_helper(locust, username):
         found_token = False
         for r_hist in response.history:
             found_token |= r_hist.cookies.get('token') is not None
-        if found_token:
-            response.success()
-            logging.debug("created user: %s", username)
-        else:
-            response.failure("login failed")
+            if found_token:
+                response.success()
+                logging.debug("created user = %s", username)
+                return found_token
+
+        response.failure("login failed")
         return found_token
```

Comment:
Could you try maybe (more-or-less equivalent code but much simpler to read):
```
for r_hist in response.history:
    if r_hist.cookies.get('token') is not None:
        response.success()
        logging.debug("created user = %s", username)
        return True
response.failure("login failed")
return False
```

Reply:
Done